### PR TITLE
Add try/except blocks around external text file read operations to catch encoding-related exceptions

### PR DESCRIPTION
**Pull request by mosherubin**
Added try/except blocks around read operations of external files. These are meant to catch any Unicode-related exceptions thrown while reading these files (e.g., the file contains an invalid UTF-8 sequence), providing more robust code. The except clauses will log a message pointing to the file that causes the exception. External files include:

- C++ source file being processed
- The filefilter.txt file
- Rule Python script files 

These modifications were made because NsiqCppStyle often failed to read external files, due to encoding issues in the source files (e.g., invalid UTF-8 sequences, incorrect UTF16 encoding).  I was often left puzzling over which file had the invalid sequence.  With the try/except blocks, the user is told which file is problematic.

### CODE REVIEW

File: nsiqcppstyle_checker.py [hunk 1]
```diff
@@ -344,7 +344,12 @@ def __init__(self, filename, data=None):
         self.data = data
         if data is None:
             f = open(filename)
```
Comment (kunaltyagi) [inline nsiqcppstyle_checker.py:346]: Let's use `with` here since the file should be closed after reading it

File: nsiqcppstyle_checker.py [hunk 2]
```diff
@@ -983,7 +988,15 @@ def Peek(self):
 
 def ProcessFile(ruleManager, file, data=None):
     #    print file
-    lexer = CppLexerNavigator(file, data)
+    try:
+        lexer = CppLexerNavigator(file, data)
+    except:
```
Comment (kunaltyagi) [inline nsiqcppstyle_checker.py:993]: Let's capture `UnicodeDecodeError` only. If there's an OSError, we don't want to skip in that case

File: nsiqcppstyle_exe.py [hunk 3]
```diff
@@ -425,7 +425,12 @@ def GetActiveFilter(self):
     def GetFilterFile(self, filterfile):
         if not os.path.exists(filterfile):
             return None
-        f = open(filterfile, 'r')
+        try:
+            f = open(filterfile, 'r')
```
Comment (kunaltyagi) [inline nsiqcppstyle_exe.py:429]: Opening a file wouldn't result in DecodeError. This error would be faced by the reader, not the `open` function

File: nsiqcppstyle_lexer.py [hunk 4]
```diff
@@ -888,6 +888,10 @@ def validate_file(self, filename):
             f = open(filename)
             lines = f.readlines()
             f.close()
```
Comment (kunaltyagi) [inline nsiqcppstyle_lexer.py:890]: Consider using a `with` outside the try

File: nsiqcppstyle_checker.py (hunk 1 above)
Comment (mosherubin) [reply nsiqcppstyle_checker.py:346]: Always a good thing - modified several other places in similar fashion.

Done.

File: nsiqcppstyle_checker.py (hunk 2 above)
Comment (mosherubin) [reply nsiqcppstyle_checker.py:993]: Done.

File: nsiqcppstyle_exe.py (hunk 3 above)
Comment (mosherubin) [reply nsiqcppstyle_exe.py:429]: Agreed, reverted the code change.

Done.

File: nsiqcppstyle_lexer.py (hunk 4 above)
Comment (mosherubin) [reply nsiqcppstyle_lexer.py:890]: This, and other file reads, now use "with".

Done.

**Comment by kunaltyagi** [comment]
I think you've not pushed the latest changes

**Comment by mosherubin** [comment]
Got it - as part of the mixup, I needed to merge my repo's pull request-created branch into its master branch.  Now you can see the changes.

File: nsiqcppstyle_lexer.py [hunk 5]
```diff
@@ -211,7 +207,17 @@ def writetab(self, tabfile, outputdir=""):
...
+    # ------------------------------------------------------------
+    # writetab() - Write lexer information to a table file
+    # ------------------------------------------------------------
+    def writetab(self, tabfile, outputdir=""):
+        if isinstance(tabfile, types.ModuleType):
+            return
+        basetabfilename = tabfile.split(".")[-1]
+        filename = os.path.join(outputdir, basetabfilename) + ".py"
+        with open(filename, "w") as tf:
+            self.actual_writetab(tabfile, tf)
```
[the remark this answers is not in the document]
Comment (mosherubin) [reply nsiqcppstyle_lexer.py]: Good programming style, thanks.

Done.

**Review by kunaltyagi — APPROVED** [review]

**Pull request merged**